### PR TITLE
Mark disable-affinity-assistant as deprecated

### PR DESCRIPTION
**Pull request by QuanZhang-William**
# Changes

This commit marks the `disable-affinity-assistant` feature flag as deprecated in favor of the new `coschedule` feature flag. See migration plan in https://github.com/tektoncd/pipeline/blob/main/docs/affinityassistants.md

Part of https://github.com/tektoncd/pipeline/issues/6988

/kind misc

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
action required: The `disable-affinity-assistant` feature flag is deprecated in favour of the new `coschedule` feature flag. The `disable-affinity-assistant` feature flag will be removed in 9 months. 
The Affinity Assistant behaviour should now be configured by the `coschedule` feature flag.
```

### CODE REVIEW

File: docs/deprecations.md [hunk 1]
```diff
@@ -27,6 +27,7 @@ The following features are deprecated but have not yet been removed.
...
 | [v1beta1 Tasks, TaskRuns, Pipelines, and PipelineRuns are deprecated in favor of v1](https://github.com/tektoncd/pipeline/issues/5541)                                                                                                                 | v0.50.0                                                              | Beta                                                                                                  | v0.62.0                       |
+| [Deprecate and remove disable-affinity-assistant feature flag](https://github.com/tektoncd/pipeline/issues/6988) in favour of the new `coschedule` configMap                                                                                                                 | v0.51.0                                                              | N/A                                                                                                  | April 27, 2024                       |
```
Comment (lbernick) [inline docs/deprecations.md:30]:
```suggestion
| [The `disable-affinity-assistant` feature flag is deprecated in favor of the `coschedule` feature flag](https://github.com/tektoncd/pipeline/issues/6988)                                                                                                                | v0.51.0                                                              | N/A                                                                                                  | April 27, 2024                       |
```

**Review by lbernick — COMMENTED** [review]
https://github.com/tektoncd/pipeline/blob/main/docs/affinityassistants.md currently states "In 9 months, the disable-affinity-assistant feature flag will be deprecated". Can you update this to clarify that the disable-affinity-assistant flag is deprecated and will be removed in 9 months? And can you please also update the other places this configuration is specified (https://github.com/tektoncd/pipeline/blob/main/docs/additional-configs.md#customizing-the-pipelines-controller-behavior, https://github.com/tektoncd/pipeline/blob/main/config/config-feature-flags.yaml) to clarify that this is deprecated?

**Comment by lbernick** [comment]
I think integration test failures are due to https://github.com/tektoncd/pipeline/issues/6992

**Comment by QuanZhang-William** [comment]
> https://github.com/tektoncd/pipeline/blob/main/docs/affinityassistants.md currently states "In 9 months, the disable-affinity-assistant feature flag will be deprecated". Can you update this to clarify that the disable-affinity-assistant flag is deprecated and will be removed in 9 months? And can you please also update the other places this configuration is specified (https://github.com/tektoncd/pipeline/blob/main/docs/additional-configs.md#customizing-the-pipelines-controller-behavior, https://github.com/tektoncd/pipeline/blob/main/config/config-feature-flags.yaml) to clarify that this is deprecated?

Thanks, clarification added

File: docs/deprecations.md (hunk 1 above)
Comment (QuanZhang-William) [reply docs/deprecations.md]: Changed

File: config/config-feature-flags.yaml [hunk 2]
```diff
@@ -29,6 +29,8 @@ data:
   # See more in the Affinity Assistant documentation
   # https://github.com/tektoncd/pipeline/blob/main/docs/affinityassistants.md
   # or https://github.com/tektoncd/pipeline/pull/2630 for more info.
+  #
+  # Note: This feature flag is deprecated and will be removed in 9 months. Consider using `coschedule` feature flag to configure Affinity Assistant behavior.
```
Comment (lbernick) [inline config/config-feature-flags.yaml:33]: Can you update line 46, where it says that coscheduling is non functional?

Comment (QuanZhang-William) [reply config/config-feature-flags.yaml:33]: Oh, sorry I thought I have removed in the last implementation PR... Removed!

**Review by lbernick — COMMENTED** [review]
Sorry for one more request-- I think having "in 9 months" is confusing in markdown files since we won't be continually updating them, can you swap to saying "in release v0.60"?

**Comment by QuanZhang-William** [comment]
> Sorry for one more request-- I think having "in 9 months" is confusing in markdown files since we won't be continually updating them, can you swap to saying "in release v0.60"?

No problem, changed. BTW, was wondering is there a place we can check future release plan?

**Review by lbernick — APPROVED** [review]

**Comment by lbernick** [comment]
> > Sorry for one more request-- I think having "in 9 months" is confusing in markdown files since we won't be continually updating them, can you swap to saying "in release v0.60"?
> 
> No problem, changed. BTW, was wondering is there a place we can check future release plan?

You can take a look at the milestones (https://github.com/tektoncd/pipeline/milestones) or the release rotation in the [pipelines wg notes](https://docs.google.com/document/d/1AfJfdyd1JN2P4haBdYOxEn6SMENMgvQF9ps7iF2QSG0/edit?resourcekey=0-lm66bE68HA_WVLA6AaCh6w#bookmark=id.2496leurrmin)

**Comment by QuanZhang-William** [comment]
/test pull-tekton-pipeline-alpha-integration-tests 
due to timeout

**Pull request merged**